### PR TITLE
Add edge support

### PR DESCRIPTION
**Pull request by davityavryan**
Hi @andregardi 

As of you are not doing any transpilation I decided just do a small change in your code which is breaking support for ALL the Edge versions. 
You can see in [MDN page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Browser_compatibility) that Spread in object literals is not supported in Edge browser. But as you can see [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Browser_compatibility) is supported from Edge 12.

### CODE REVIEW

**Comment by davityavryan** [comment]
I can also setup environment so that you will have automatic transpiled code to browsers defined in `.browserslist` if you wish (with separate PR, it's juts current problem is critical for our company at the moment).

File: index.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 function setState(store, newState, afterUpdateCallback) {
-  store.state = { ...store.state, ...newState };
+  store.state = Object.assign(store.state, newState);
```
Comment (davityavryan) [inline index.js:2]: @andregardi  and. just FYI: one to one fix would have been Object.assign({}, store.state, newState); but I decided to mutate state partially. Hit me if you see problem there.

**Comment by davityavryan** [comment]
@andregardi Hello?

**Comment by andregardi** [comment]
Thank you for that!
Sorry for the delay. I am going to make some more tests and add to the next release.

**Comment by davityavryan** [comment]
Thanks but you just reverted my changes with commit https://github.com/andregardi/use-global-hook/commit/ee2ece9926347cf937f9a2d9406a8d95a55c9373

**Comment by andregardi** [comment]
> Thanks but you just reverted my changes with commit [ee2ece9](https://github.com/andregardi/use-global-hook/commit/ee2ece9926347cf937f9a2d9406a8d95a55c9373)

Hello @davityavryan 
Unfortunately your commit didn't work on the counters sandbox example. Some counter did not updated the value after I merged your changes. Triggering the listeners is the most important part of this lib.

I also tested the new versions of Edge and it seems to support spread operators now. I also do not love the idea to give up on latests ES features to make it compatible with legacy browsers. Because it will be always used with React, the developer can always config babel to transpile the code to the target client.
I hope that doesn't demotivate you. Thank you for your contribution.

**Comment by davityavryan** [comment]
Hey @andregardi 

It does not. Thanks for comment. I might try to find something else suiting better for my needs. No problem.

## Just as a notes

I have commented under this PR which might have had helped you with the problem running the counters sandbox example. I think it was because hook was not seeing any change as of object reference was not updating. I was expecting that you will agree to change to the exact same code...
image

And regarding babel transpilation, nobody transpiles `node_modules`, but here is a hint for all ppl who will come here and ask for help. 
This is a fix for older browsers so that you will also transpile `use-blobal-hook` if you need to for Webpack.
image

**Comment by andregardi** [comment]
Actually you are right @davityavryan !
We should have a modern ES source code but the main exported could be a transpiled version in ES5 export.
We had PR for that one. But it also had other changes, not retrocompatible with previews versions and we didn't merged it.

Thank you for bringing this up.

**Comment by davityavryan** [comment]
If you need help with that, I can help by adding rollup to repo. which will produce all necessary versions.

**Comment by andregardi** [comment]
I actually never heard about rollup before. I only know Babel and Webpack.
But if that works for creating an ES5 export, your help is very welcome!

**Comment by davityavryan** [comment]
See #50
